cmds/fwtools/flash/flash_linux.go

```diff
@@ -155,21 +155,17 @@ func run(args []string, supportedProgrammers map[string]programmerInit) (reterr
 			return err
```

Comment:
Reading can be:

```
if err := io.Copy(f, io.NewSectionReader(programmer, 0, programmer.Size()); err != nil {
  return err
}
```

This is probably a bit faster (write and read operations happen concurrently). Also, it is a friendlier on the RAM (no RAM buffer).

Reply:
great idea. it's in the planning, but I will be adding size and offset as flags, they're very handy.